pipeline_components/aiplatform/aiplatform/custom_train_job/component.py

```diff
@@ -0,0 +1,167 @@
+
+    ```
+    train_xgb_model.py \
+        --train_data <train_data.path> \
+        --valid_data <valid_data.path> \
+        --test_data <test_data.path> \
+        --metrics <metrics.path> \
+        --hparams json.dumps(<hparams>)
+    ```
+
+    Ensure that your task can read these arguments and outputs metrics to the provided
+    path and the model to the correct path based on:
+    https://cloud.google.com/vertex-ai/docs/training/code-requirements.
+
+    Args:
+        task_uri (str): gs:// uri to python task script. See:
+            https://cloud.google.com/vertex-ai/docs/training/code-requirements.
+        train_data (Dataset): Training data (passed as an argument to task script)
+        valid_data (Dataset): Validation data (passed as an argument to task script)
+        test_data (Dataset): Test data (passed as an argument to task script).
+        staging_bucket (str): Staging bucket for CustomTrainingJob.
+        project_location (str): location of the Google Cloud project.
+        project_id (str): project id of the Google Cloud project.
+        model_display_name (str): Name of the new trained model version.
+        train_container_uri (str): Container URI for running task script.
+        serving_container_uri (str): Container URI for deploying the output model.
+        model (Model): Trained model output.
+        metrics (Metrics): Output metrics of trained model.
+        requirements (List[str]): Additional python dependencies for training script.
+        job_name (str): Name of training job.
+        hparams (Dict[str, str]): Hyperparameters (passed as a JSON serialised argument
+            to task script)
+        replica_count (int): Number of replicas (increase for distributed tasks).
+        machine_type (string): Machine type of compute.
+        accelerator_type (string): Accelerator type (change for GPU support).
+        accelerator_count (string): Accelerator count (increase for GPU cores).
+
+    Returns:
+        parent_model (str): Resource URI of the parent model (empty string if the
+            trained model is the first model version of its kind).
+    """
+    import json
+    import logging
+    import os.path
+    import time
+    import google.cloud.aiplatform as aip
+
+    logging.info(f"Checking if parent model with name {model_display_name} exists")
+    models = aip.Model.list(filter=f"displayName={model_display_name}")
+
+    # check if a model with the same name already exists
+    if len(models) == 0:
+        logging.info("No parent model found.")
+        parent_model = None
+        is_default_version = True
+    elif len(models) == 1:
+        parent_model = models[0].resource_name
+        is_default_version = False
+        logging.info(f"Parent model found: {parent_model}")
+    else:
+        raise RuntimeError(
+            f"Multiple models with name {model_display_name} were found."
+        )
+
```

Comment:
we already have a component that looks up an existing model that we are using elsewhere in the pipelines. I suggest we remove this logic from the training component and instead accept Model artifact as an input. WDYT?